server/routes/api/users.js

```diff
@@ -0,0 +1,78 @@
+const express = require("express");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const router = express.Router();
+
+const User = require("../../models/user");
+const secret = require("../../config/config").appSecret;
+
+const {
+  validateRegisterInput,
+  validateLoginInput
+} = require("../../validation/users");
+
+router.post("/register", (req, res) => {
+  const { errors, isValid } = validateRegisterInput(req.body);
+
+  if (!isValid) {
+    res.status(400).json(errors);
+  } else {
+    //Validation passed
+    const { firstName, lastName, email, password } = req.body;
```

Comment:
validatation/users.js expects a confirmPassword field, but you're not including it.

Maybe you're waiting until you add the field to the front-end?

Reply:
The `confirmPassword` field is only needed for validation, once validation has passed we don't need it for the actual registration process, so it's not extracted from `req.body`

Reply:
You're right, that was me reading your code wrong!